features/keychain/module/crypto/secp256k1.js

```diff
@@ -1,22 +1,40 @@
 import assert from 'minimalistic-assert'
 import elliptic from '@exodus/elliptic'
 import { mapValues, pick } from '@exodus/basic-utils'
+import ecc from '@exodus/bitcoinerlab-secp256k1'
 
-const validEcOptions = (ecOptions) =>
+import { tweakPrivateKey, tweakPublicKey } from './tweak'
+
+const isValidEcOptions = (ecOptions) =>
   !ecOptions || Object.keys(ecOptions).every((key) => ['canonical'].includes(key))
 
+const isValidTweakOptions = (tweakOptions) =>
+  !tweakOptions ||
+  Object.keys(tweakOptions).every((key) => ['tweakHash', 'extraEntropy'].includes(key))
+
 export const create = ({ getPrivateHDKey }) => {
   const EC = elliptic.ec
   const curve = new EC('secp256k1')
 
   const createInstance = () => ({
     signBuffer: async ({ seedId, keyId, data, ecOptions, enc = 'der' }) => {
       assert(['der', 'raw'].includes(enc), 'signBuffer: invalid enc')
-      assert(validEcOptions(ecOptions), 'signBuffer: invalid EC option')
+      assert(isValidEcOptions(ecOptions), 'signBuffer: invalid EC option')
       const { privateKey } = getPrivateHDKey({ seedId, keyId })
       const signature = curve.sign(data, privateKey, pick(ecOptions, ['canonical']))
       return enc === 'der' ? Buffer.from(signature.toDER()) : { ...signature }
     },
+    signSchnorr: async ({ seedId, keyId, data, tweak = false, tweakOptions }) => {
```

Comment:
Any reason why we don't create a new signer from the tweaked private key like this:
```typescript
const tweakedSigner = signer.addTweak(tweak)
tweakedSigner.signBuffer(...)
await tweakedSigner.getPublicKey()
etc...
```

Reply:
How would we use that over RPC?

Reply:
> How would we use that over RPC?

How would we use any of these signers over the RPC? I think in general that was overlooked on my part :| I'm okay with keeping the API as proposed in this PR though.

Reply:
The other ones are not problematic, because they take simple types as args and return simple types as results. `signer.addTweak(tweak)` would however return functions. We would need to somehow call it in the keychain and keep the result (the new signer) in the keychain. And yet know how to call the new signer of RPC.

Reply:
I like the concept though.

Reply:
I see, Diego has rectified my mistakes already, I had already forgot 😅  Let's keep the current API. I'm wondering if it would be nicer if we could aggregate `tweak` & `tweakOptions` into one object to keep the API a bit cleaner?

Reply:
My initial implementation was to use the presence of `tweakOptions` as `tweak == true` and I was certain that I would be asked to change it :grin:

Reply:
https://github.com/ExodusMovement/exodus-oss/pull/90/commits/482aa2454b35ad4cc8e474fdb23ef56784278383

Reply:
Hmm. The `extraEntropy` doesn't really belong in `tweak.options`.